app/Http/Controllers/DashboardController.php

```diff
@@ -2,13 +2,40 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Casts\SubTask;
+use App\Models\Course;
+use App\Models\Enums\CorrectionType;
+use App\Models\Enums\TaskTypeEnum;
+use App\Models\Grade;
+use Carbon\Carbon;
 use Illuminate\Http\Request;
 use Illuminate\View\View;
+use App\Models\Task;
+use App\Models\User;
 
 class DashboardController extends Controller
 {
-    public function index() : View
+    public function index(): View
     {
-        return view('dashboard');
+        $courses = auth()->user()->courses()->get();
+        $tasks = Task::whereIn('course_id', $courses->pluck('id'))->where('ends_at', '>=', now())->assignments()->orderBy('ends_at', 'asc')->get();
+        $nextAssignment = $tasks->first();
+        $courseAssignments = Task::assignments()->whereIn('course_id', $courses->pluck('id'))->get();
+        $exercises = Task::exercises()->whereIn('course_id', $courses->pluck('id'))->orderBy('starts_at', 'asc')->limit(5)->get();
+        $completedAssignments = Grade::where('value', '=', 'passed')->get();
```

Comment:
There is no reason to pull all grades here. Only fetch relevant ones.

Reply:
That is actually from when we had the progress bars, I will just delete it.